tests/prove_and_verify.rs

```diff
@@ -0,0 +1,151 @@
+use plonky::{verify_proof_circuit, CircuitBuilder, Curve, Field, PartialWitness, Tweedledee, Tweedledum};
+use std::time::Instant;
+
+// Make sure it's the same as in `plonk.rs`.
+const NUM_WIRES: usize = 9;
+
+#[test]
+fn test_proof_trivial_circuit() {
+    let mut builder = CircuitBuilder::<Tweedledee>::new(128);
+    let t = builder.constant_wire(<Tweedledee as Curve>::ScalarField::ZERO);
+    builder.assert_zero(t);
+    let mut partial_witness = PartialWitness::new();
+    partial_witness.set_target(t, <Tweedledee as Curve>::ScalarField::ZERO);
+    let circuit = builder.build();
+    let witness = circuit.generate_witness(partial_witness);
+    let proof = circuit.generate_proof::<Tweedledum>(witness, true).unwrap();
+    assert!(verify_proof_circuit::<Tweedledee, Tweedledum>(&[], &proof, &circuit,).is_ok());
+}
+
+#[test]
+fn test_proof_sum_circuit() {
+    let mut builder = CircuitBuilder::<Tweedledee>::new(128);
+    let t1 = builder.constant_wire(<Tweedledee as Curve>::ScalarField::ZERO);
+    let t2 = builder.constant_wire(<Tweedledee as Curve>::ScalarField::ZERO);
+    let s = builder.add(t1, t2);
+    builder.assert_zero(s);
+    let mut partial_witness = PartialWitness::new();
+    partial_witness.set_target(t1, <Tweedledee as Curve>::ScalarField::ZERO);
+    partial_witness.set_target(t2, <Tweedledee as Curve>::ScalarField::ZERO);
+    let circuit = builder.build();
+    let witness = circuit.generate_witness(partial_witness);
+    let proof = circuit.generate_proof::<Tweedledum>(witness, true).unwrap();
+    assert!(verify_proof_circuit::<Tweedledee, Tweedledum>(&[], &proof, &circuit,).is_ok());
+}
+
+#[test]
+fn test_proof_sum_big_circuit() {
+    let now = Instant::now();
+    let mut builder = CircuitBuilder::<Tweedledee>::new(128);
+    let ts = (0..10_000)
```

Comment:
Don't need to change anything now, but I think it would be nice if `cargo test` only ran fast tests by default, so we could run that regularly. Not sure how best to handle large tests -- we could `#[ignore]` them, or use a cfg rule, or make them binaries.

Reply:
I `#[ignore]`'d all the big tests.